packages/tool-cache/src/tool-cache.ts

```diff
@@ -201,7 +201,13 @@ export async function extractTar(
 
   dest = dest || (await _createExtractFolder(dest))
   const tarPath: string = await io.which('tar', true)
-  await exec(`"${tarPath}"`, [flags, '-C', dest, '-f', file])
+  const args: string[] = [flags]
+  if (IS_WINDOWS) {
+    // On Windows, absolute path use a ':' (e.g. C:\foo), which is incorrectly recognized as a remote path by tar unless this flag is passed
+    args.push('--force-local')
+  }
+  args.push('-C', dest, '-f', file)
```

Comment:
According to `C:\windows\system32\tar.exe --help` the option `-C` is only for create, it is not a common option.

Also @joshmgross informed me bsd tar does not support `--force-local`. And `C:\windows\system32\tar.exe` is bsd tar.

Instead of `-C` I think dest should be the working directory (exec option).

Thoughts?

Reply:
[ExecOptions.cwd](https://github.com/actions/toolkit/blob/master/packages/exec/src/interfaces.ts#L7)

Reply:
@ericsciple, while using option `cwd` works, I am not sure about `--force-local`. It seems to be required. See:

- https://github.com/eine/setup-msys2/commit/4278cd9b1b01d0d7fc0a0cc42a0b2769d44fb5f3/checks?check_suite_id=346924107
- https://github.com/eine/setup-msys2/commit/cd6ad53aa10f2461e6affd7cdc81715eca390f4d/checks?check_suite_id=346927420
- https://github.com/eine/setup-msys2/compare/cwd...eine:cwd-noforce

Reply:
Oh I see. Because of the other file path :facepalm:

The problem is the hosted machines prepend the gnu tools to the PATH, rather than append to the PATH.

Self hosted runners would more commonly have %windir%\System32\tar.exe (bsd tar) in the PATH.

And bsd tar works correctly. It also doesn't require forward slashes.

I think the correct solution is: on Windows don't which `tar`.

Instead do:
```javascript
`${process.env['windir']}\\System32\\tar.exe`
```

If tar doesn't exist at that location, fail.

Thoughts?

Reply:
@ericsciple, thanks for the details. If that is the case, I think that more effort is required than just fixing `if (IS_WINDOWS)` in this repo:

- The toolkit should provide a `tar` command that would hide the inconsistencies in the implementation of the runners from the user. Not using `which` has the side effect that the arguments might need to be explicitly different for windows-latest, ubuntu-latest and macos-latest.
- Your comment about gnu tools being prepended to the default (bsd flavour), should added to the docs as an important warning.

Then, it will be up to action developers to use tha command or to handle it with custom logic.

Reply:
I dont like the solution i proposed anymore. @thboop good point that it doesnt cover older windows which the runner supports.